src/metabase/models/humanization.clj

```diff
@@ -16,75 +19,111 @@
+                     (capitalize-word part))))))
+
+;; :none is just an identity implementation
+(defmethod name->human-readable-name :none
+  ([s]   s)
+  ([_ s] s))
 
 
 (defn- custom-display-name?
-  "Is DISPLAY-NAME a custom name that was set manually by the user in the metadata edit screen?"
+  "Is `display-name` a custom name that was set manually by the user in the metadata edit screen?"
   [internal-name display-name]
-  (and (not= display-name (name->human-readable-name:simple   internal-name))
-       (not= display-name (name->human-readable-name:advanced internal-name))))
-
-(defn- re-humanize-names! [model]
-  (doseq [{id :id, internal-name :name, display-name :display_name} (db/select [model :id :name :display_name])
-          :let                                                      [new-display-name (name->human-readable-name internal-name)]
-          :when                                                     (and (not= display-name new-display-name)
-                                                                         (not (custom-display-name? internal-name display-name)))]
-    (log/info (format "Updating display name for %s '%s': '%s' -> '%s'" (name model) internal-name display-name new-display-name))
-    (db/update! model id
-      :display_name new-display-name)))
+  ;; Try all the different implementations of name->human-readable-name. If any (f internal-name) is equal to
+  ;; display-name we can safely assume that display-name was set automatically using that strategy and stop there.
+  ;; Otherwise if none of the strategies match (i.e. `some` returns `nil`) we can assume a custom name was set.
+  (nil? (some #(= display-name (% internal-name))
+              (vals (methods name->human-readable-name)))))
+
+(defn- re-humanize-names!
+  "Update all non-custom display names of all instances of `model` (e.g. Table or Field). To prevent explosions on very
+  large instances this is done in chunks of 10000 IDs at a time"
+  ;; TODO - once Toucan has the new streaming SELECT functions we can use those instead of manually chunking
+  ;; TODO - Alternatively, just make this chunking pattern generic and add to Toucan. Could be useful elsewhere!
+  ([model] (re-humanize-names! model 0))
+  ([model starting-id]
+   (let [chunk-size 10000]
+     (when-let [instances (seq (db/select [model :id :name :display_name]
+                                 :id [:>= starting-id]
+                                 {:limit chunk-size}))]
+       (doseq [{id :id, internal-name :name, display-name :display_name} instances
+               :let  [new-display-name (name->human-readable-name internal-name)]
+               :when (and (not= display-name new-display-name)
+                          (not (custom-display-name? internal-name display-name)))]
+         (log/info (format "Updating display name for %s '%s': '%s' -> '%s'"
+                           (name model) internal-name display-name new-display-name))
+         (db/update! model id
```

Comment:
I think as it sits now, this could be pretty slow on larger schemas. Seems like we have some options to improve it. I think we currently use autocommit. With that I think the above statement will commit on each update. Switching to a transaction would help there. Another possibility, which is more work, but something that might be worth it in the future is exposing batch statements through toucan.

Reply:
the speed as which it gets done isn't really a big concern here as long as it gets done at some point. I added the chunking stuff mainly to avoid using too much memory. The previous logic pulled down all of the records at once which seemed sketchy